### ISSUE TITLE
Can't install pglogical for postgresql 9.6 (Depends: postgresql-9.6 (>= 9.6.10) but 9.6.5-1.pgdg80+1 is to be installed)

### ISSUE THREAD

**Issue by cpl-cpi** [2ndQuadrant/pglogical#400]
I get this error:
```
sudo apt-get install postgresql-9.6-pglogical
Reading package lists... Done
Building dependency tree
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 postgresql-9.6-pglogical : Depends: postgresql-9.6 (>= 9.6.10) but 9.6.5-1.pgdg80+1 is to be installed
E: Unable to correct problems, you have held broken packages.
```
As it's a production database I want to change as less as possible in the system...

What do you recommend in my case?

Thanks!

**Comment by bonesmoses** [comment]
The situation is very clear and has little to do with the pglogical codebase itself. The version of pglogical you are trying to install depends on Postgres 9.6.10 or higher, which you are refusing to install. Either determine a time where you can install 9.6.10 or greater, or compile pglogical yourself.

It is also my duty to inform you that 9.6.10 was released over 4 years ago, and 9.6.24 is the latest. 9.6.5 was released over 5 years ago. You are missing several crash, corruption, security, and other bug fixes from the last 5 years of releases, in addition to not having compatibility to newer compiled libraries and extensions. Which if these is a greater risk to your production database? A small amount of downtime to update to the latest patch, or several security vulnerabilities and other dangerous bugs?

**Comment by cpl-cpi** [comment]
First of all: thanks for your detailed answer!

> The situation is very clear and has little to do with the pglogical codebase itself. The version of pglogical you are trying to install depends on Postgres 9.6.10 or higher, which you are refusing to install. Either determine a time where you can install 9.6.10 or greater, or compile pglogical yourself.

So If I understand you right, there is no ready to install package for my PostgreSQL version (9.6.5-1.pgdg80+1), correct?

> It is also my duty to inform you that 9.6.10 was released over 4 years ago, and 9.6.24 is the latest. 9.6.5 was released over 5 years ago. You are missing several crash, corruption, security, and other bug fixes from the last 5 years of releases, in addition to not having compatibility to newer compiled libraries and extensions.

Yes. In fact it's my customer who has the risk as it's his database. But he want's to migrate from AWS to GCP (that is why we install PGLogical).

> Which if these is a greater risk to your production database? A small amount of downtime to update to the latest patch, or several security vulnerabilities and other dangerous bugs?

So I'm with you but the question is:
Does it make sense to upgrade the database just for migration when the PGLogical extension can be installed (somehow) and the migration can be done? It's all about getting the Data save from A to B and not securing the old database.

The migration with PGLogical as a dependency of the GCP database migration service works fine (even with old database / PGLogical version). The only struggle is to install PGLogical.

**Comment by luss** [comment]
It pretty definitely makes sense to do a minor upgrade to your old database
to the latest 9.6.x.  It's a simple upgrade.  It's only scary cuz you
haven't been doing it on a regular basis (at least once a year or better
yet quarterly).  GCP will do this for you in the future...
